### DISCUSSION TITLE
aws-amplify-alpha - how to specify platform: WEB_COMPUTE?

### DISCUSSION THREAD

**Discussion by FeroHriadel** [aws/aws-cdk#31123] [Q&A]
I just installed @aws-cdk/aws-amplify-alpha@2.39.1-alpha.0. I want to deploy a nextjs app with SSR. I need to specify:

```
const amplifyApp = new amplify.App(this, 'MyApp', {
  platform: amplify.Platform.WEB_COMPUTE,
});
```

It's also in the docs. But it says it's an unknown property. What should I do now?

**Comment by khushail** [comment]
Hi @FeroHriadel  , the Platform type is available in [CDK docs](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-amplify-alpha.Platform.html) and I can deploy it with this code below -
First install the alpha module by running this command -

```
 sudo npm install @aws-cdk/aws-amplify-alpha@2.166.0-alpha.0
```
```ts

    const amplifyapp = new amplify.App(this, 'amplifyapp', {
      platform: amplify.Platform.WEB_COMPUTE,
    });

    new cdk.CfnOutput(this, 'amplifyappurl', {
        value: amplifyapp.appId,
    });
```

Synthesized template -
```json
  "amplifyapp996AC4E0": {
   "Type": "AWS::Amplify::App",
   "Properties": {
    "BasicAuthConfig": {
     "EnableBasicAuth": false
    },
    "IAMServiceRole": {
     "Fn::GetAtt": [
      "amplifyappRole19C50DA6",
      "Arn"
     ]
    },
    "Name": "amplifyapp",
    "Platform": "WEB_COMPUTE"
   },
   "Metadata": {
    "aws:cdk:path": "AmplifyIssueStack/amplifyapp/Resource"
   }
```

and the deployment succeeds -

<img width="961" alt="Screenshot 2024-11-14 at 1 48 55 PM" src="https://github.com/user-attachments/assets/712ab27b-c9a7-4fdc-b316-47bdd074eb85">


Hope this would be helpful. Let me know if this does not work.
